source/SIL.AppBuilder.Portal.Frontend/src/data/containers/resources/organization/with-data-actions.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+
+import { defaultOptions, OrganizationResource, ProductDefinitionResource } from '@data';
+import { recordIdentityFromKeys } from '@data/store-helpers';
+import { requireProps } from '@lib/debug';
+import { withData as withOrbit, WithDataProps } from 'react-orbitjs';
+import { OrganizationAttributes } from '@data/models/organization';
+
+export interface IProvidedProps {
+  updateAttribute: (attribute: string, value: any) => Promise<any>;
+  updateAttributes: (attrs: OrganizationAttributes) => any;
+  updateProductDefinition: (productDefinition: ProductDefinitionResource) => any;
+}
+
+interface IOwnProps {
+  organization: OrganizationResource;
+}
+
+
+type IProps =
+  & IOwnProps
+  & WithDataProps;
+
+export function withDataActions<T>(WrappedComponent) {
+
+  class OrganizationDataActionWrapper extends React.Component<IProps & T> {
+
+    updateAttribute = async (attribute: string, value: any) => {
+      const { organization, dataStore } = this.props;
+
+      await dataStore.update(
+        q => q.replaceAttribute(organization, attribute, value),
+        defaultOptions()
+      );
+
+      this.forceUpdate();
+    }
+
+    updateAttributes = (attributes: OrganizationAttributes) => {
+      const { organization, dataStore } = this.props;
+      const { id, type } = organization;
+
+      return dataStore.update(q => q.replaceRecord({
+        id, type, attributes
+      }), defaultOptions());
+    }
+
+    updateProductDefinition = (productDefinition) => {
+
+      const { organization, dataStore } = this.props;
+      const recordIdentity = recordIdentityFromKeys(organization);
+
+      // TODO: Handle organizationRelationShip
```

Comment:
todo?

Reply:
Yeah I'm working on it... just added to don't miss it.